### ISSUE TITLE
The cookie expiration time "" is not valid.

### ISSUE THREAD

**Issue by noxify** [FriendsOfPHP/Goutte#259]
Hi guys,

i have a problem on my remote server, but i don't know why.
On my local environment, everything is working fine.

Local ENV:
- OSX 10.11.3
- MAMP PRO ( Virtual Host has PHP 5.6.10 active)

Remote ENV:
- Shared Hosting ( Unix - df.eu )
- PHP 5.6.12

My Code snippet ( simplified without my special logic ):
```
public function checkLink()
	{
		$product = 'http://store.nike.com/de/de_de/pd/air-max-bw-ultra-herrenschuh/pid-10872127/pgid-11161768';

		try {

			$client = new \Goutte\Client();

			$guzzleClient = new \GuzzleHttp\Client( [
				'curl' => [
					CURLOPT_SSL_VERIFYPEER  => false,
					CURLOPT_SSL_VERIFYHOST  => false
				],
				'defaults' => [
					'cookies'   => false
				],
				'debug' => true
			] );

			$client->setClient( $guzzleClient );

			$crawler = $client->request( 'GET', $product );

		} catch( \Exception $ex) {

			\Log::error($ex->getMessage());

		}

		\Log::info($client->getResponse()->getStatus());
		\Log::info($client->getResponse()->getHeaders());

		return view('welcome');
	}
```

Debug Print - local ENV:
```
* Hostname in DNS cache was stale, zapped
*   Trying 95.101.72.167...
* Connected to store.nike.com (95.101.72.167) port 80 (#0)
> GET /de/de_de/pd/air-max-bw-ultra-herrenschuh/pid-10872127/pgid-11161768 HTTP/1.1
host: store.nike.com
user-agent: Symfony2 BrowserKit

< HTTP/1.1 200 OK
< Server: Jetty(9.2.4.v20141028)
< X-Request-Id: 955122
< Content-Type: text/html; charset=UTF-8
< Content-Language: de-DE
< X-Nike-Build-Rev: bb694cac35771b096053d6f065f8c06e542f09ca
< Cache-Control: max-age=2084
< Date: Sun, 06 Mar 2016 13:37:18 GMT
< Transfer-Encoding:  chunked
< Connection: keep-alive
< Connection: Transfer-Encoding
< Set-Cookie: AnalysisUserId=95.101.72.163.44101457271438570; expires=Mon, 31-Dec-2038 23:59:59 GMT; path=/; domain=.nike.com
< 

* Connection #0 to host store.nike.com left intact
```

Debug Print - Remote ENV:
```
* Hostname was NOT found in DNS cache
*   Trying 2.20.188.164...
* Connected to store.nike.com (2.20.188.164) port 80 (#0)
> GET /de/de_de/pd/air-max-bw-ultra-herrenschuh/pid-10872127/pgid-11161768 HTTP/1.1
user-agent: Symfony2 BrowserKit
host: store.nike.com

< HTTP/1.1 200 OK
* Server Jetty(9.2.4.v20141028) is not blacklisted
< Server: Jetty(9.2.4.v20141028)
< X-Request-Id: 661066
< Content-Type: text/html; charset=UTF-8
< Content-Language: de-DE
< X-Nike-Build-Rev: bb694cac35771b096053d6f065f8c06e542f09ca
< X-EdgeConnect-MidMile-RTT: 0
< X-EdgeConnect-Origin-MEX-Latency: 315
< X-EdgeConnect-MidMile-RTT: 6
< X-EdgeConnect-Origin-MEX-Latency: 315
< Cache-Control: max-age=2045
< Date: Sun, 06 Mar 2016 13:38:52 GMT
< Transfer-Encoding:  chunked
< Connection: keep-alive
< Connection: Transfer-Encoding
< Set-Cookie: AnalysisUserId=2.20.188.160.7071457271532666; expires=Mon, 31-Dec-2038 23:59:59 GMT; path=/; domain=.nike.com
< 
* Connection #0 to host store.nike.com left intact
```

On my local env it looks that everything works fine.
On my remote server, the code is running into the catch statement...

```
[2016-03-06 15:03:21] local.ERROR: exception 'UnexpectedValueException' with message 'The cookie expiration time "" is not valid.' in /<serverpath>/vendor/symfony/browser-kit/Cookie.php:76
```

Last but not least, here the require parts from my composer.json
```
    "require": {
        "php": ">=5.5.9",
        "laravel/framework": "5.2.*",
        "barryvdh/laravel-ide-helper": "^2.1",
        "barryvdh/laravel-debugbar": "^2.1",
        "pseudoagentur/soa-sentinel": "dev-develop",
        "laravelcollective/html": "5.2.x-dev",
        "greggilbert/recaptcha": "2.1.0",
        "cartalyst/stripe": "~2.0",
        "cartalyst/stripe-laravel": "4.0.*",
        "caffeinated/menus": "dev-master",
        "barryvdh/laravel-dompdf": "0.6.*",
        "doctrine/dbal": "^2.5",
        "thomaswelton/laravel-gravatar": "dev-master",
        "proengsoft/laravel-jsvalidation": "v1.3.1",
        "filp/whoops": "2.0.0",
        "arcanedev/log-viewer": "3.*",
        "fabpot/goutte": "^3.1"
    },
    "require-dev": {
        "fzaninotto/faker": "~1.4",
        "mockery/mockery": "0.9.*",
        "phpunit/phpunit": "~4.0",
        "symfony/css-selector": "2.8.*|3.0.*",
        "symfony/dom-crawler": "2.8.*|3.0.*"
    },
```

Composer told me the following in detail:

```
installed:
  arcanedev/log-viewer                  3.7.9               Provides a Log Viewer for Laravel 5
  arcanedev/support                     3.7.5               ARCANEDEV Support Helpers
  barryvdh/elfinder-flysystem-driver    v0.1.4              A Flysystem Driver for elFinder
  barryvdh/laravel-debugbar             v2.2.0              PHP Debugbar integration for Laravel
  barryvdh/laravel-dompdf               v0.6.1              A DOMPDF Wrapper for Laravel
  barryvdh/laravel-ide-helper           v2.1.4              Laravel IDE Helper, generates correct PHPDocs for all Facade classes, to improve auto-completion.
  caffeinated/menus                     dev-master dc396a8  Laravel 5 Menus
  cartalyst/collections                 v1.1.0              Collection Abstaction library for PHP.
  cartalyst/sentinel                    v2.0.9              PHP 5.4+ Fully-featured Authentication & Authorization System
  cartalyst/stripe                      v2.0.3              A comprehensive Stripe API package.
  cartalyst/stripe-laravel              v4.0.0              Laravel 5.2 integration for the Cartalyst Stripe package.
  cartalyst/support                     v1.1.2              Support helpers.
  classpreloader/classpreloader         3.0.0               Helps class loading performance by generating a single PHP file containing all of the autoloaded files for a specific use case
  dimsav/laravel-translatable           v5.4                A Laravel package for multilingual models
  dnoegel/php-xdg-base-dir              0.1                 implementation of xdg base directory specification for php
  doctrine/annotations                  v1.2.7              Docblock Annotations Parser
  doctrine/cache                        v1.6.0              Caching library offering an object-oriented API for many cache backends
  doctrine/collections                  v1.3.0              Collections Abstraction library
  doctrine/common                       v2.6.1              Common Library for Doctrine projects
  doctrine/dbal                         v2.5.4              Database Abstraction Layer
  doctrine/inflector                    v1.1.0              Common String Manipulations with regard to casing and singular/plural rules.
  doctrine/instantiator                 1.0.5               A small, lightweight utility to instantiate objects in PHP without invoking their constructors
  doctrine/lexer                        v1.0.1              Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
  dompdf/dompdf                         v0.6.2              DOMPDF is a CSS 2.1 compliant HTML to PDF converter
  fabpot/goutte                         v3.1.2              A simple PHP Web Scraper
  filp/whoops                           2.0.0               php error handling for cool kids
  fzaninotto/faker                      v1.5.0              Faker is a PHP library that generates fake data for you.
  greggilbert/recaptcha                 2.1.0               reCAPTCHA Validator for Laravel 5
  guzzlehttp/guzzle                     6.1.1               Guzzle is a PHP HTTP client library
  guzzlehttp/promises                   1.0.3               Guzzle promises library
  guzzlehttp/psr7                       1.2.3               PSR-7 message implementation
  hamcrest/hamcrest-php                 v1.2.2              This is the PHP port of Hamcrest Matchers
  intervention/image                    2.3.6               Image handling and manipulation library with support for Laravel integration
  jakub-onderka/php-console-color       0.1                 
  jakub-onderka/php-console-highlighter v0.3.2              
  jeremeamia/SuperClosure               2.2.0               Serialize Closure objects, including their context and binding
  laravel/framework                     v5.2.22             The Laravel Framework.
  laravelcollective/html                5.2.x-dev 02c75df   HTML and Form Builders for the Laravel Framework
  league/flysystem                      1.0.17              Filesystem abstraction: Many filesystems, one API.
  maximebf/debugbar                     v1.11.1             Debug bar in the browser for php application
  mockery/mockery                       0.9.4               Mockery is a simple yet flexible PHP mock object framework for use in unit testing with PHPUnit, PHPSpec or any other testin...
  monolog/monolog                       1.18.0              Sends your logs to files, sockets, inboxes, databases and various web services
  mtdowling/cron-expression             v1.1.0              CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
  nesbot/carbon                         1.21.0              A simple API extension for DateTime.
  nikic/php-parser                      v2.0.1              A PHP parser written in PHP
  paragonie/random_compat               v1.2.1              PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
  phenx/php-font-lib                    0.2.2               A library to read, parse, export and make subsets of different types of font files.
  phpdocumentor/reflection-docblock     2.0.4               
  phpspec/prophecy                      v1.6.0              Highly opinionated mocking framework for PHP 5.3+
  phpunit/php-code-coverage             2.2.4               Library that provides collection, processing, and rendering functionality for PHP code coverage information.
  phpunit/php-file-iterator             1.4.1               FilterIterator implementation that filters files based on a list of suffixes.
  phpunit/php-text-template             1.2.1               Simple template engine.
  phpunit/php-timer                     1.0.7               Utility class for timing
  phpunit/php-token-stream              1.4.8               Wrapper around PHP's tokenizer extension.
  phpunit/phpunit                       4.8.23              The PHP Unit Testing framework.
  phpunit/phpunit-mock-objects          2.3.8               Mock Object library for PHPUnit
  proengsoft/laravel-jsvalidation       v1.3.1              Validate forms transparently with Javascript reusing your Laravel Validation Rules, Messages, and FormRequest
  pseudoagentur/soa-elfinder            3.0.0               A Laravel Package to integrate elFinder 2
  pseudoagentur/soa-flash               3.0.0               Easy flash notifications
  pseudoagentur/soa-jsvalidation        3.0.0               Validate forms transparently with Javascript reusing your Laravel Validation Rules, Messages, and FormRequest
  pseudoagentur/soa-sentinel            dev-develop d176de0 Administrative interface builder for Laravel with Sentinel Integration
  pseudoagentur/user-settings           3.0.0               Simple user settings facade for Laravel 5.
  psr/http-message                      1.0                 Common interface for HTTP messages
  psr/log                               1.0.0               Common interface for logging libraries
  psy/psysh                             v0.7.1              An interactive shell for modern PHP.
  sebastian/comparator                  1.2.0               Provides the functionality to compare PHP values for equality
  sebastian/diff                        1.4.1               Diff implementation
  sebastian/environment                 1.3.5               Provides functionality to handle HHVM/PHP environments
  sebastian/exporter                    1.2.1               Provides the functionality to export PHP variables for visualization
  sebastian/global-state                1.1.1               Snapshotting of global state
  sebastian/recursion-context           1.0.2               Provides functionality to recursively process PHP variables
  sebastian/version                     1.0.6               Library that helps with managing the version number of Git-hosted PHP projects
  studio-42/elfinder                    2.1.7               File manager for web
  swiftmailer/swiftmailer               v5.4.1              Swiftmailer, free feature-rich PHP mailer
  symfony/browser-kit                   v3.0.3              Symfony BrowserKit Component
  symfony/class-loader                  v3.0.3              Symfony ClassLoader Component
  symfony/console                       v3.0.3              Symfony Console Component
  symfony/css-selector                  v3.0.3              Symfony CssSelector Component
  symfony/debug                         v3.0.3              Symfony Debug Component
  symfony/dom-crawler                   v3.0.3              Symfony DomCrawler Component
  symfony/event-dispatcher              v3.0.3              Symfony EventDispatcher Component
  symfony/finder                        v3.0.3              Symfony Finder Component
  symfony/http-foundation               v3.0.3              Symfony HttpFoundation Component
  symfony/http-kernel                   v3.0.3              Symfony HttpKernel Component
  symfony/polyfill-mbstring             v1.1.0              Symfony polyfill for the Mbstring extension
  symfony/polyfill-php56                v1.1.0              Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
  symfony/polyfill-util                 v1.1.0              Symfony utilities for portability of PHP codes
  symfony/process                       v3.0.3              Symfony Process Component
  symfony/routing                       v3.0.3              Symfony Routing Component
  symfony/translation                   v3.0.3              Symfony Translation Component
  symfony/var-dumper                    v3.0.3              Symfony mechanism for exploring and dumping PHP variables
  symfony/yaml                          v3.0.3              Symfony Yaml Component
  thomaswelton/gravatarlib              0.1.0               A lightweight PHP 5.3 OOP library providing easy gravatar integration.
  thomaswelton/laravel-gravatar         dev-master b9eeb9c  Laravel 5 Gravatar helper
  vlucas/phpdotenv                      v2.2.0              Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.
```

Maybe you have an idea to solve this problem.

Thanks a lot!

**Comment by noxify** [comment]
I think i found the problem: https://github.com/symfony/symfony/issues/15488 

Tested a bit around and saw that i got "false" as response in the parseDate() function on my remote server.


YIPPPIII.. so it's not a Goutte/Guzzle Problem.

**Comment by buttererror** [comment]
I use windows 32 bit on my windows tablet, and had this problem, I found out that is a php issue with the year 2038 as the link provided above would explain why as it is. After searching, I installed editthiscookie extension for chrome, I found two cookie, i tried to modify them but after refresh especially with laravel session, everything return as it is after refresh. After digging, in session.php, I altered the 'lifetime' property from 20 * 365 * 24 * 60 to 10 * 365 * 24 * 60, as 10 refers to the number of years. ah, the problem with editthiscookie that it can't modify the cookie, I read that I could remove any '__' character in the dirver name in session.php, but i didn't give it a try
